test_rmw_implementation/test/test_subscription.cpp

```diff
@@ -377,3 +377,56 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_with_bad_args) {
+  bool taken = false;
+  test_msgs__msg__BasicTypes output_message{};
```

Comment:
@Lobotuerk nit: it'd be nice to not use the default initialization i.e. pick a couple fields and change them.

Reply:
Updated